src/helpers/validation/userRegister.ts

```diff
@@ -0,0 +1,38 @@
+import * as Yup from 'yup';
+
+const userRegisterSchema = Yup.object().shape({
+  role: Yup.string().required('Role is required').oneOf(['user', 'therapist'], 'Role must be either user or therapist'),
+  fullName: Yup.string().required('Full name is required'),
+  email: Yup.string().email('Invalid email address').required('Email is required'),
+  password: Yup.string()
+    .required('Password is required')
+    .min(8, 'Password must be at least 8 characters long'),
+  confirmPassword: Yup.string().required('Confirm password is required').oneOf([Yup.ref('password')], 'Passwords must match'),
```

Comment:
most of the time no need to send it to the server it's a client validation process to make sure that the user entered the intended password